src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface_kernels.jl

```diff
@@ -150,81 +151,100 @@ using Printf
+                                     Gᵁ, Gⱽ, Hᶠᶜ, Hᶜᶠ, g, 
+                                     timestepper)
+    k_top = grid.Nz+1
+    
     TX, TY, _ = topology(grid)
 
     @inbounds begin 
-        advance_previous_velocity!(i, j, 1, timestepper, U, Uᵐ⁻¹, Uᵐ⁻²)
-        advance_previous_velocity!(i, j, 1, timestepper, V, Vᵐ⁻¹, Vᵐ⁻²)
+        advance_previous_velocity!(i, j, k_top-1, timestepper, U, Uᵐ⁻¹, Uᵐ⁻²)
+        advance_previous_velocity!(i, j, k_top-1, timestepper, V, Vᵐ⁻¹, Vᵐ⁻²)
 
         # ∂τ(U) = - ∇η + G
-        U[i, j, 1] +=  Δτ * (- g * Hᶠᶜ[i, j] * ∂xᶠᶜᶠ_η(i, j, k_top, grid, TX, η★, timestepper, η, ηᵐ, ηᵐ⁻¹, ηᵐ⁻²) + Gᵁ[i, j, 1])
-        V[i, j, 1] +=  Δτ * (- g * Hᶜᶠ[i, j] * ∂yᶜᶠᶠ_η(i, j, k_top, grid, TY, η★, timestepper, η, ηᵐ, ηᵐ⁻¹, ηᵐ⁻²) + Gⱽ[i, j, 1])
-
+        U[i, j, k_top-1] +=  Δτ * (- g * Hᶠᶜ[i, j, 1] * ∂xᶠᶜᶠ_η(i, j, k_top, grid, TX, η★, timestepper, η, ηᵐ, ηᵐ⁻¹, ηᵐ⁻²) + Gᵁ[i, j, k_top-1])
+        V[i, j, k_top-1] +=  Δτ * (- g * Hᶜᶠ[i, j, 1] * ∂yᶜᶠᶠ_η(i, j, k_top, grid, TY, η★, timestepper, η, ηᵐ, ηᵐ⁻¹, ηᵐ⁻²) + Gⱽ[i, j, k_top-1])
+                          
         # time-averaging
-        η̅[i, j, k_top] += averaging_weight * η[i, j, k_top]
-        U̅[i, j, 1]     += averaging_weight * U[i, j, 1]
-        V̅[i, j, 1]     += averaging_weight * V[i, j, 1]
+        η̅[i, j, k_top]   += averaging_weight * η[i, j, k_top]
+        U̅[i, j, k_top-1] += averaging_weight * U[i, j, k_top-1]
+        V̅[i, j, k_top-1] += averaging_weight * V[i, j, k_top-1]
     end
 end
 
-function split_explicit_free_surface_substep!(η, state, auxiliary, settings, weights, arch, grid, g, Δτ, substep_index)
-    # unpack state quantities, parameters and forcing terms 
-    U, V             = state.U,    state.V
-    Uᵐ⁻¹, Uᵐ⁻²       = state.Uᵐ⁻¹, state.Uᵐ⁻²
-    Vᵐ⁻¹, Vᵐ⁻²       = state.Vᵐ⁻¹, state.Vᵐ⁻²
-    ηᵐ, ηᵐ⁻¹, ηᵐ⁻²   = state.ηᵐ,   state.ηᵐ⁻¹, state.ηᵐ⁻²
-    η̅, U̅, V̅          = state.η̅, state.U̅, state.V̅
-    Gᵁ, Gⱽ, Hᶠᶜ, Hᶜᶠ = auxiliary.Gᵁ, auxiliary.Gⱽ, auxiliary.Hᶠᶜ, auxiliary.Hᶜᶠ
-
-    timestepper      = settings.timestepper
-    averaging_weight = weights[substep_index]
-
-    parameters = auxiliary.kernel_parameters
-
-    args = (grid, Δτ, η, ηᵐ, ηᵐ⁻¹, ηᵐ⁻², U, V, Uᵐ⁻¹, Uᵐ⁻², Vᵐ⁻¹, Vᵐ⁻², 
-            η̅, U̅, V̅, averaging_weight, 
-            Gᵁ, Gⱽ, g, Hᶠᶜ, Hᶜᶠ, timestepper)
+# Barotropic Model Kernels
+# u_Δz = u * Δz
+@kernel function _barotropic_mode_kernel!(U, V, grid, u, v)
+    i, j  = @index(Global, NTuple)	
+    k_top = grid.Nz+1
 
-    launch!(arch, grid, parameters, split_explicit_free_surface_evolution_kernel!,        args...)
-    launch!(arch, grid, parameters, split_explicit_barotropic_velocity_evolution_kernel!, args...)
+    # hand unroll first loop
+    @inbounds U[i, j, k_top-1] = Δzᶠᶜᶜ(i, j, 1, grid) * u[i, j, 1]
+    @inbounds V[i, j, k_top-1] = Δzᶜᶠᶜ(i, j, 1, grid) * v[i, j, 1]
 
-    return nothing
+    @unroll for k in 2:grid.Nz
```

Comment:
seems like this PR re-introduced some unrolling...!

it's obviously a mistake and it stems from the fact that we don't require PRs to be updated with `main` before merging... 

It's a bit scary that somethings we do may be undone by some other PR later without nobody noticing. To avoid that I'll turn on the requirement that PRs are up to date with main before merging.

@glwagner, @simone-silvestri, @tomchor, @siddharthabishnu

Reply:
sounds good, go for it